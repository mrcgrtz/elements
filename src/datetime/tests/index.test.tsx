/**
 * DateTime Test
 * @author Marc Görtz <https://marcgoertz.de/>
 */

import React from 'react';
import { render } from '@testing-library/react';
import DateTime from '..';

describe('<DateTime />', () => {
  it('renders a date and time information', () => {
    const mockDate = new Date(1999, 3, 17, 22, 11, 18);
    const { container } = render(<DateTime dateTime={mockDate} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  describe('on different times of the day', () => {
    [...Array(24).keys()].forEach((hour) => {
      it(`renders a text for ${hour} o’clock`, () => {
        const mockDate = new Date(2020, 1, 12, hour, 0, 0);
        const { container } = render(<DateTime dateTime={mockDate} />);
        expect(container.textContent).toMatchSnapshot();
      });
    });
  });
});
