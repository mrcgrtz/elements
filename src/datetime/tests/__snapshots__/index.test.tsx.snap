// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DateTime /> on different times of the day renders a text for 0 o’clock 1`] = `"00:00 UhrNachts am 12. Februar 2020"`;

exports[`<DateTime /> on different times of the day renders a text for 1 o’clock 1`] = `"01:00 UhrNachts am 12. Februar 2020"`;

exports[`<DateTime /> on different times of the day renders a text for 2 o’clock 1`] = `"02:00 UhrNachts am 12. Februar 2020"`;

exports[`<DateTime /> on different times of the day renders a text for 3 o’clock 1`] = `"03:00 UhrNachts am 12. Februar 2020"`;

exports[`<DateTime /> on different times of the day renders a text for 4 o’clock 1`] = `"04:00 UhrVerdammt früh am Morgen des 12. Februar 2020"`;

exports[`<DateTime /> on different times of the day renders a text for 5 o’clock 1`] = `"05:00 UhrVerdammt früh am Morgen des 12. Februar 2020"`;

exports[`<DateTime /> on different times of the day renders a text for 6 o’clock 1`] = `"06:00 UhrVerdammt früh am Morgen des 12. Februar 2020"`;

exports[`<DateTime /> on different times of the day renders a text for 7 o’clock 1`] = `"07:00 UhrMorgens am 12. Februar 2020"`;

exports[`<DateTime /> on different times of the day renders a text for 8 o’clock 1`] = `"08:00 UhrMorgens am 12. Februar 2020"`;

exports[`<DateTime /> on different times of the day renders a text for 9 o’clock 1`] = `"09:00 UhrMorgens am 12. Februar 2020"`;

exports[`<DateTime /> on different times of the day renders a text for 10 o’clock 1`] = `"10:00 UhrAm frühen Vormittag des 12. Februar 2020"`;

exports[`<DateTime /> on different times of the day renders a text for 11 o’clock 1`] = `"11:00 UhrVormittags am 12. Februar 2020"`;

exports[`<DateTime /> on different times of the day renders a text for 12 o’clock 1`] = `"12:00 UhrZur Mittagszeit am 12. Februar 2020"`;

exports[`<DateTime /> on different times of the day renders a text for 13 o’clock 1`] = `"13:00 UhrZur Mittagszeit am 12. Februar 2020"`;

exports[`<DateTime /> on different times of the day renders a text for 14 o’clock 1`] = `"14:00 UhrAm frühen Nachmittag des 12. Februar 2020"`;

exports[`<DateTime /> on different times of the day renders a text for 15 o’clock 1`] = `"15:00 UhrNachmittags am 12. Februar 2020"`;

exports[`<DateTime /> on different times of the day renders a text for 16 o’clock 1`] = `"16:00 UhrNachmittags am 12. Februar 2020"`;

exports[`<DateTime /> on different times of the day renders a text for 17 o’clock 1`] = `"17:00 UhrAm späten Nachmittag des 12. Februar 2020"`;

exports[`<DateTime /> on different times of the day renders a text for 18 o’clock 1`] = `"18:00 UhrAm frühen Abend des 12. Februar 2020"`;

exports[`<DateTime /> on different times of the day renders a text for 19 o’clock 1`] = `"19:00 UhrAm frühen Abend des 12. Februar 2020"`;

exports[`<DateTime /> on different times of the day renders a text for 20 o’clock 1`] = `"20:00 UhrAbends am 12. Februar 2020"`;

exports[`<DateTime /> on different times of the day renders a text for 21 o’clock 1`] = `"21:00 UhrAbends am 12. Februar 2020"`;

exports[`<DateTime /> on different times of the day renders a text for 22 o’clock 1`] = `"22:00 UhrSpätabends am 12. Februar 2020"`;

exports[`<DateTime /> on different times of the day renders a text for 23 o’clock 1`] = `"23:00 UhrKurz vor Mitternacht am 12. Februar 2020"`;

exports[`<DateTime /> renders a date and time information 1`] = `
.c0 {
  font-family: system-ui,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto, Helvetica,Arial,sans-serif,'Apple Color Emoji','Segoe UI Emoji', 'Segoe UI Symbol';
  font-size: 0.75rem;
  line-height: 1.5;
  color: dimgrey;
  white-space: nowrap;
}

.c1 {
  margin-right: 0.75ex;
  width: 1em;
  height: 1em;
  vertical-align: -2px;
  fill: currentColor;
}

<time
  class="c0"
  datetime="1999-04-17T22:11:18"
  title="Samstag, der 17. April 1999 um 22:11 Uhr"
>
  <svg
    aria-hidden="true"
    class="c1"
    height="12"
    viewBox="0 0 12 12"
    width="12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      22:11
       Uhr
    </title>
    <circle
      cx="6"
      cy="6"
      fill="none"
      r="5.425"
      stroke="currentColor"
      stroke-width="1.15"
    />
    <line
      fill="none"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-width="1.35"
      transform="rotate(660 6 6)"
      x1="6"
      x2="6"
      y1="6"
      y2="4"
    />
    <line
      fill="none"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-width="1.35"
      transform="rotate(66 6 6)"
      x1="6"
      x2="6"
      y1="6"
      y2="2.5"
    />
  </svg>
  Spätabends am
   
  17. April 1999
</time>
`;
